db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/StoreDataManagerProvider.java

```diff
@@ -15,42 +15,37 @@
  */
 package org.commonjava.indy.infinispan.data;
 
-import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.cassandra.data.ClusterStoreDataManager;
+import org.commonjava.indy.core.conf.IndyDurableStateConfig;
 import org.commonjava.indy.data.StandaloneStoreDataManager;
 import org.commonjava.indy.data.StoreDataManager;
 
-import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.inject.Default;
 import javax.enterprise.inject.Produces;
 import javax.inject.Inject;
 
 @ApplicationScoped
-public class StoreDataManagerProvider {
+public class StoreDataManagerProvider
+{
 
     @Inject
-    @StandaloneStoreDataManager
-    InfinispanStoreDataManager infinispanStoreDataManager;
-
-    @Inject
-    IndyConfiguration indyConfiguration;
-
-    private StoreDataManager storeDataManager;
-
-    private boolean isStandaloneMode;
-
-    @PostConstruct
-    public void init()
-    {
-        isStandaloneMode = indyConfiguration.isStoreManagerStandalone();
-    }
+    IndyDurableStateConfig durableStateConfig;
 
     @Produces
     @Default
-    public StoreDataManager getStoreDataManager()
+    public StoreDataManager getStoreDataManager(
```

Comment:
imo, we should not inject both but only one. This may break (maybe sometime in future, under some circumstances) where only one impl can be initiated.

Reply:
I was trying to implement this by construct the specific implementation according to the config, but there are too many @Inject variables in the class as well as the abstract one, which means we may need to inject them first in the provider first, then passing them as construct parameters. I believe the ideal way for this case is setting them @Default and @Alternative respectively, and declare the one that we will use in the bean.xml. But we need to support to switch it according to the config, so I think this is not a bad way so far, see Gjorgji's previous ref in the following doc: https://docs.jboss.org/weld/reference/2.4.6.Final/en-US/pdf/weld-reference.pdf (page 29). WDYT ?

Reply:
The alternative must be in beans.xml which is inside of our code. in our case, it is not convenient to switch. 
If we use same tarball plus config, the ideal way is to create an impl via constructor. But .... as you said there are many injections and not convinient to construct. 
Then the last way is to stick to the current code. Once we get it stable, we can update the code to use only one. I am fine with it.

Reply:
See, thanks.